src/java.base/share/classes/java/lang/invoke/BootstrapMethodInvoker.java

```diff
@@ -145,6 +145,10 @@ else if (info.getClass() == int[].class) {
                 } else if (isLambdaMetafactoryAltMetafactoryBSM(bsmType)) {
                     maybeReBoxElements(argv);
                     result = (CallSite)bootstrapMethod.invokeExact(caller, name, (MethodType)type, argv);
+                } else if (isObjectMethodsBootstrapBSM(bsmType)) {
+                    MethodHandle[] methodHandles = new MethodHandle[argv.length - 2];
+                    System.arraycopy(argv, 2, methodHandles, 0, argv.length - 2);
```

Comment:
Are we avoiding `Arrays.copyOfRange(argv, 2, argv.length, MethodHandle.class)` because of the JIT? If JIT compiler can recognize `MethodHandle.class` constant and inline I recommend using copyOfRange like for string concat above.

Reply:
No, I actually reached for the appropriate `copyOfRange` method, but my IDE hid it from view in favor of `copyOfRange(T[], int, int)` and all the others.